common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -260,67 +260,69 @@ public static CharSequence escapeCsv(CharSequence value) {
      */
     public static CharSequence escapeCsv(CharSequence value, boolean trimWhiteSpace) {
         int length = checkNotNull(value, "value").length();
-        if (length == 0) {
-            return value;
-        }
-
-        int start = 0;
-        int last = length - 1;
-        boolean trimmed = false;
+        int start;
+        int last;
         if (trimWhiteSpace) {
             start = indexOfFirstNonOwsChar(value, length);
-            if (start == length) {
-                return EMPTY_STRING;
-            }
             last = indexOfLastNonOwsChar(value, start, length);
-            trimmed = start > 0 || last < length - 1;
-            if (trimmed) {
-                length = last - start + 1;
+        } else {
+            start = 0;
+            last = length - 1;
+        }
+        if (start > last) {
+            return EMPTY_STRING;
+        }
+
+        int firstUnescapedSpecial = -1;
+        boolean quoted = false;
+        if (isDoubleQuote(value.charAt(start))) {
+            quoted = isDoubleQuote(value.charAt(last)) && last > start;
```

Comment:
have you considered checking `last > start` first? I would assume this check would be cheaper and may allow a short circuit.